packages/expo-manifests/ios/EXManifests/EXManifestsManifest.swift

```diff
@@ -0,0 +1,359 @@
+//  Copyright © 2021 650 Industries. All rights reserved.
+
+// this uses abstract class patterns
+// swiftlint:disable unavailable_function
+
+import Foundation
+import UIKit
+
+/**
+ Uses objective-c NSExceptions for field validation. This is to maintain
+ backwards compatibility with the previous objective-c implementation so that we don't need to do
+ error handling at every callsite. When all the code is swift, we might be able to use these exceptions.
+ */
+
+extension Dictionary where Key == String {
+  func optionalValue<T>(forKey: String) -> T? {
+    guard let value = self[forKey] else {
+      return nil
+    }
+
+    if !(value is T) {
+      let exception = NSException(
+        name: NSExceptionName.internalInconsistencyException,
+        reason: "Value for (key = %@) has incorrect type",
+        userInfo: ["key": forKey]
+      )
+      exception.raise()
+    }
+    return value as? T
+  }
+
+  func requiredValue<T>(forKey: String) -> T {
+    let value = self[forKey]
+
+    if value == nil {
+      let exception = NSException(
+        name: NSExceptionName.internalInconsistencyException,
+        reason: "Value for (key = %@) is null",
+        userInfo: ["key": forKey]
+      )
+      exception.raise()
+    }
+
+    if !(value is T) {
+      let exception = NSException(
+        name: NSExceptionName.internalInconsistencyException,
+        reason: "Value for (key = %@) has incorrect type",
+        userInfo: ["key": forKey]
+      )
+      exception.raise()
+    }
+
+    // exception above will preempt force_cast
+    // swiftlint:disable:next force_cast
+    return value as! T
```

Comment:
Then, since we guarded that `value` is `T`, there is no need to force cast 😉

```suggestion
    return value
```